crates/turbo-tasks/src/manager.rs

```diff
@@ -382,59 +394,60 @@ impl<B: Backend> TurboTasks<B> {
 
         let this = self.pin();
         let future = async move {
-            loop {
-                if this.stopped.load(Ordering::Acquire) {
-                    break;
-                }
-                if let Some(execution) = this.backend.try_start_task_execution(task_id, &*this) {
-                    // Setup thread locals
-                    let (result, duration, instant) = CELL_COUNTERS
-                        .scope(Default::default(), async {
-                            let (result, duration, instant) =
-                                TimedFuture::new(AssertUnwindSafe(execution.future).catch_unwind())
-                                    .await;
-                            (result, duration, instant)
-                        })
-                        .await;
-                    if cfg!(feature = "log_function_stats") && duration.as_millis() > 1000 {
-                        println!(
-                            "{} took {}",
-                            this.backend.get_task_description(task_id),
-                            FormatDuration(duration)
-                        )
+            #[allow(clippy::blocks_in_if_conditions)]
+            while CURRENT_TASK_STATE
+                .scope(Default::default(), async {
```

Comment:
Could this return a [ControlFlow](https://doc.rust-lang.org/stable/std/ops/enum.ControlFlow.html) instead, just to make it clear that we're breaking or continuing a loop?